test/specs/products.js

```diff
@@ -0,0 +1,163 @@
+        await ProductsPage.logout();
+    });
+
+
+    it(`should have the '${URLS.products}' url`, async () => {
+
+        expect(await LoginPage.browserUrl()).toEqual(URLS.products);
+    });
+
+    it('should have the "SWAGLABS" logo in header', async () => {
+
+        expect(await ProductsPage.appLogo.isDisplayed()).toBe(true);
+    });
+
+    it('should have the Burger Menu button in header', async () => {
+
+        expect(await ProductsPage.btnBurgerMenu.isDisplayed()).toBe(true);
+    });
+
+    it('should have the Cart Icon in header', async () => {
+
+        expect(await ProductsPage.ShoppingCartIcon.isDisplayed()).toBe(true);
+    });
+
+    //#TODO: sa continui cu testele de UI de aici
+})
+
+describe("Products Test", () => {
+
+    beforeEach(async function () {
+        await LoginPage.open();
+        await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+    });
+
+    afterEach(async function () {
+        await ProductsPage.logout();
+    });
+
+    it(`should have the '${URLS.products}' url`, async () => {
+
+        expect(await LoginPage.browserUrl()).toEqual(URLS.products);
+    });
+
+    it("the shopping cart label number should increase after adding a product", async () => {
+        await ProductsPage.btnAddBackpack.click();
+
+        expect(await ProductsPage.ShoppingCartLabel.getText()).toEqual("1");
+        await ProductsPage.btnRemoveBackpack.click();
+    });
+
+    it("the shopping cart label number should decrease after removing a product", async () => {
+        await ProductsPage.addMultipleItems();
+        await ProductsPage.btnRemoveBackpack.click();
+
+        expect(await ProductsPage.ShoppingCartLabel.getText()).toEqual("1");
+        await ProductsPage.btnRemoveBikeLight.click();
+    });
+
+    it("the shopping cart label should not be displayed upon removing the last item", async () => {
+        await ProductsPage.btnAddBackpack.click();
+        await ProductsPage.btnRemoveBackpack.click();
+
+        expect(await ProductsPage.ShoppingCartLabel.isDisplayed()).toBe(false);
+    });
+
+    it("should display all the 6 products names", async () => {
+        const productsNamesText = await ProductsPage.productNames();
+
+        expect(productsNamesText).toContain(ProductNames.backpack);
+        expect(productsNamesText).toContain(ProductNames.bikeLight);
+        expect(productsNamesText).toContain(ProductNames.boltShirt);
+        expect(productsNamesText).toContain(ProductNames.fleeceJacket);
+        expect(productsNamesText).toContain(ProductNames.onesie);
+        expect(productsNamesText).toContain(ProductNames.redShirt);
+    });
+
+    it("should display all the 6 products images", async () => {
+        let productsImagesSource = [];
+
+        const productImage = await ProductsPage.allItemsImages;
+        for (let i = 0; i < productImage.length; i++) {
+            productsImagesSource.push(await productImage[i].getAttribute("src"));
+        }
```

Comment:
maybe try to create a method for extracting images, names and prices.

Reply:
the method "productData()" has been created in the ProductPage file